### PR TITLE
flags: Take -a and -p from environment as default

### PR DESCRIPTION
**Pull request by camh-**
Take the values for the flags `-a` and `-p` from environment variables
as the default, still allowing them to be overridden on the command
line.

Sometimes using `servedir`, you may always want to listen on all
interfaces but the default invocation of `servedir` in a Makefile recipe
does not specify `-a`. You can now set `SERVEDIR_ALL_INTERFACES=true` in
the environment and these pre-canned `servedir` invocations can work
with your defaults.

This switches the use of the `flag` package to use an explicit `FlagSet`
instead of using the global one in the `flag` package. Using the global
one makes it a lot harder to test as flags get redefined.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -2,16 +2,15 @@
 // the current directory or a specified directory, on the next free
 // ephemeral port or a specified port.
 //
-//     go get foxygo.at/servedir
-//     servedir --help
-//     usage: servedir [-a] [-p <port>] [<dir>]
+//	go run foxygo.at/servedir@latest --help
+//	usage: servedir [-a] [-p <port>] [<dir>]
```
Comment (juliaogris) [inline main.go:6]: seeing this makes me want to have `gofmt` or `gofumpt` as part of the build process, not as part of this pr. just noting.

Arguably the reformat and help wording improvement could be part of a different commit?

**Review by juliaogris — APPROVED** [review]
Nicely done! I like this a lot. Makes me wonder if I should get remove kong as a dependency from evy....

**Review by juliaogris — APPROVED** [review]
Nicely done! Makes me wonder if I should try and remove the kong dependency from evy. 💚

File: main.go (hunk 1 above)
Comment (camh-) [reply main.go:6]: Since I was re-doing the `go get` to `go run`, it seemed fine to do the other reformatting in this commit. Furthermore since the flags are re-worked in this commit, reworking the comments with the flags seems to fit to me too. It's all flags :flags: \</wave-hand>

**Comment by camh-** [comment]
> Makes me wonder if I should try and remove the kong dependency from evy

:lollipop: As I was doing it, I was wondering if I should just add kong as a dependency here :joy:
